Added verbose logging, added minor improvements

garimellagowtham opened the pull request:
1. Add proto option for goal tolerance
2. Fix bug in landed state not accepting internal transition
3. Added logs folder where glog files are stored
4. Add comments to protof files
5. Fix multithread takeoff test to make test deterministic
6. Add options to simulator.launch to add GLOG options

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/jhu-asco/aerial_autonomy/30)
<!-- Reviewable:end -->

msheckells commented:
Reviewed 14 of 14 files at r1.
Review status: all files reviewed at latest revision, 6 unresolved discussions, some commit checks failed.

---

*[include/aerial_autonomy/actions_guards/position_control_functors.h, line 95 at r1](https://reviewable.io:443/reviews/jhu-asco/aerial_autonomy/30#-Kf3ip2Cor72lO8jK68D:-Kf3ip2Cor72lO8jK68E:b22qzh9) ([raw file](https://github.com/jhu-asco/aerial_autonomy/blob/bd5867e3930160c07229445dcac86c8924457353/include/aerial_autonomy/actions_guards/position_control_functors.h#L95)):*
> ```objc
>     double yaw = data.rpydata.z;
>     // Define tolerance and check if reached goal
>     // TODO Use a parameter for setting position tolerance
> ```

Remove todo

---

*[include/aerial_autonomy/state_machines/basic_state_machine.h, line 215 at r1](https://reviewable.io:443/reviews/jhu-asco/aerial_autonomy/30#-Kf3khrNVxTBjc9IIcMy:-Kf3khrNVxTBjc9IIcMz:b-shjzjf) ([raw file](https://github.com/jhu-asco/aerial_autonomy/blob/bd5867e3930160c07229445dcac86c8924457353/include/aerial_autonomy/state_machines/basic_state_machine.h#L215)):*
> ```objc
>   void no_transition(Event const &e, FSM &, int state_index) {
>     no_transition_event_index_ = typeid(e);
>     LOG(WARNING) << "Event " << no_transition_event_index_.name()
> ```

I assume this doesn't get called for internal transitions (which don't really trigger transitions)?

---

*[include/aerial_autonomy/state_machines/state_machine_gui_connector.h, line 65 at r1](https://reviewable.io:443/reviews/jhu-asco/aerial_autonomy/30#-Kf3kVDqD2XKaC8A6aOH:-Kf3kVDqD2XKaC8A6aOI:baxvo0s) ([raw file](https://github.com/jhu-asco/aerial_autonomy/blob/bd5867e3930160c07229445dcac86c8924457353/include/aerial_autonomy/state_machines/state_machine_gui_connector.h#L65)):*
> ```objc
>   */
>   void eventCallback(const std_msgs::StringConstPtr &event_data) {
>     VLOG(2) << "Triggering event: " << event_data;
> ```

why level 2 instead of 1 here?  

---

*[include/aerial_autonomy/state_machines/state_machine_gui_connector.h, line 80 at r1](https://reviewable.io:443/reviews/jhu-asco/aerial_autonomy/30#-Kf3kYrbYHMeMkq2AsyU:-Kf3kYrbYHMeMkq2AsyV:bmsmkvo) ([raw file](https://github.com/jhu-asco/aerial_autonomy/blob/bd5867e3930160c07229445dcac86c8924457353/include/aerial_autonomy/state_machines/state_machine_gui_connector.h#L80)):*
> ```objc
>     pose_command.z = pose.pose.position.z;
>     pose_command.yaw = tf::getYaw(pose.pose.orientation);
>     VLOG(2) << "Received Pose command: " << pose_command.x << "\t"
> ```

same question as above

---

*[logs/.gitignore, line 1 at r1](https://reviewable.io:443/reviews/jhu-asco/aerial_autonomy/30#-Kf3mI7pZksxEI9jpcXX:-Kf3mI7pZksxEI9jpcXY:bpfb875) ([raw file](https://github.com/jhu-asco/aerial_autonomy/blob/bd5867e3930160c07229445dcac86c8924457353/logs/.gitignore#L1)):*
> ```
> *
> ```

Probably don't want to ignore README 

---

*[tests/state_machines/basic_state_machine_tests.cpp, line 197 at r1](https://reviewable.io:443/reviews/jhu-asco/aerial_autonomy/30#-Kf3m8abBg65uPMfYRo9:-Kf3m8abBg65uPMfYRoA:b-nxhime) ([raw file](https://github.com/jhu-asco/aerial_autonomy/blob/bd5867e3930160c07229445dcac86c8924457353/tests/state_machines/basic_state_machine_tests.cpp#L197)):*
> ```cpp
> 
>   void takeoffEventCall() {
>     for (int count = 0; count < 200; ++count) {
> ```

I was just wondering why Takeoff needs to be called 200 times

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/jhu-asco/aerial_autonomy/30)*
<!-- Sent from Reviewable.io -->

garimellagowtham commented:
Review status: 10 of 15 files reviewed at latest revision, 6 unresolved discussions.

---

*[include/aerial_autonomy/actions_guards/position_control_functors.h, line 95 at r1](https://reviewable.io:443/reviews/jhu-asco/aerial_autonomy/30#-Kf3ip2Cor72lO8jK68D:-Kf5E85OKDSA21dm_ddP:b-896fix) ([raw file](https://github.com/jhu-asco/aerial_autonomy/blob/2637cf38f6331200ba001c2978f7798678e1b4bc/include/aerial_autonomy/actions_guards/position_control_functors.h#L95)):*
<details><summary><i>Previously, msheckells (Matt Sheckells) wrote…</i></summary><blockquote>

Remove todo
</blockquote></details>

Done.

---

*[include/aerial_autonomy/state_machines/basic_state_machine.h, line 215 at r1](https://reviewable.io:443/reviews/jhu-asco/aerial_autonomy/30#-Kf3khrNVxTBjc9IIcMy:-Kf4rFz7bkk2CfJX2v4V:b-4mz7wv) ([raw file](https://github.com/jhu-asco/aerial_autonomy/blob/2637cf38f6331200ba001c2978f7798678e1b4bc/include/aerial_autonomy/state_machines/basic_state_machine.h#L215)):*
<details><summary><i>Previously, msheckells (Matt Sheckells) wrote…</i></summary><blockquote>

I assume this doesn't get called for internal transitions (which don't really trigger transitions)?
</blockquote></details>

This function gets called for any events that do not have an entry in the transition table in state machine or states. The internal transitions are accepted by states. So it will not trigger this function. The transition can be from the current state to the same state again.

---

*[include/aerial_autonomy/state_machines/state_machine_gui_connector.h, line 65 at r1](https://reviewable.io:443/reviews/jhu-asco/aerial_autonomy/30#-Kf3kVDqD2XKaC8A6aOH:-Kf4rqRRLkGBM2BM1mBE:b-886gsx) ([raw file](https://github.com/jhu-asco/aerial_autonomy/blob/2637cf38f6331200ba001c2978f7798678e1b4bc/include/aerial_autonomy/state_machines/state_machine_gui_connector.h#L65)):*
<details><summary><i>Previously, msheckells (Matt Sheckells) wrote…</i></summary><blockquote>

why level 2 instead of 1 here?  
</blockquote></details>

Fixed

---

*[include/aerial_autonomy/state_machines/state_machine_gui_connector.h, line 80 at r1](https://reviewable.io:443/reviews/jhu-asco/aerial_autonomy/30#-Kf3kYrbYHMeMkq2AsyU:-Kf4rxghirzI1wAv17Ry:by73ew9) ([raw file](https://github.com/jhu-asco/aerial_autonomy/blob/2637cf38f6331200ba001c2978f7798678e1b4bc/include/aerial_autonomy/state_machines/state_machine_gui_connector.h#L80)):*
<details><summary><i>Previously, msheckells (Matt Sheckells) wrote…</i></summary><blockquote>

same question as above
</blockquote></details>

Fixed. I was just trying to check, if different verbose levels work.

---

*[logs/.gitignore, line 1 at r1](https://reviewable.io:443/reviews/jhu-asco/aerial_autonomy/30#-Kf3mI7pZksxEI9jpcXX:-Kf4s6801IJct660kyCn:b-s6h2e0) ([raw file](https://github.com/jhu-asco/aerial_autonomy/blob/2637cf38f6331200ba001c2978f7798678e1b4bc/logs/.gitignore#L1)):*
<details><summary><i>Previously, msheckells (Matt Sheckells) wrote…</i></summary><blockquote>

Probably don't want to ignore README 
</blockquote></details>

Added lines to negate README and gitignore

---

*[tests/state_machines/basic_state_machine_tests.cpp, line 197 at r1](https://reviewable.io:443/reviews/jhu-asco/aerial_autonomy/30#-Kf3m8abBg65uPMfYRo9:-Kf4sgKJh3ILDiFqFSeQ:bkxodo0) ([raw file](https://github.com/jhu-asco/aerial_autonomy/blob/2637cf38f6331200ba001c2978f7798678e1b4bc/tests/state_machines/basic_state_machine_tests.cpp#L197)):*
<details><summary><i>Previously, msheckells (Matt Sheckells) wrote…</i></summary><blockquote>

I was just wondering why Takeoff needs to be called 200 times
</blockquote></details>

Changed to calling takeoff once and internal transition 200 times

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/jhu-asco/aerial_autonomy/30)*
<!-- Sent from Reviewable.io -->

msheckells commented:
Reviewed 5 of 5 files at r2.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/jhu-asco/aerial_autonomy/30)*
<!-- Sent from Reviewable.io -->
